src/bson.c

```diff
@@ -1114,8 +1115,13 @@ PHP_FUNCTION(toJSON)
 		RETVAL_STRINGL(str, str_len, 1);
 		bson_free(str);
 	} else {
-		RETURN_NULL();
+		phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Could not read document from BSON reader");
 	}
+
+	if (bson_reader_read(reader, &eof) || !eof) {
+		phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Reading document did not exhaust input buffer");
```

Comment:
This will yield in two exceptions being thrown because the former has no "return"